lavalink/node.py

```diff
@@ -226,34 +233,30 @@ async def connect(self, timeout=None):
             Time after which to timeout on attempting to connect to the Lavalink websocket,
             ``None`` is considered never, but the underlying code may stop trying past a
             certain point.
-
+        shutdown: bool
+            Whether the node was told to shut down
         Raises
         ------
         asyncio.TimeoutError
             If the websocket failed to connect after the given time.
+        AbortingConnectionException:
+            If the connection attempt must be aborted during a reconnect attempt
         """
-        self._is_shutdown = False
+        self._is_shutdown = shutdown
         if self.secured:
             uri = f"wss://{self.host}:{self.port}"
         else:
             uri = f"ws://{self.host}:{self.port}"
 
         ws_ll_log.info("Lavalink WS connecting to %s with headers %s", uri, self.headers)
-
-        await asyncio.wait_for(self._multi_try_connect(uri), timeout)
-
-        ws_ll_log.debug("Creating Lavalink WS listener.")
-        if self._listener_task is not None:
-            self._listener_task.cancel()
-        self._listener_task = self.loop.create_task(self.listener())
-        self._listener_task.add_done_callback(task_callback_exception)
-        self.loop.create_task(self._configure_resume()).add_done_callback(task_callback_debug)
-        if self._queue:
-            for data in self._queue:
-                await self.send(data)
-            self._queue.clear()
-        self._ready_event.set()
-        self.update_state(NodeState.READY)
+        if self.try_connect_task is not None:
+            self.try_connect_task.cancel()
+        try:
+            self.try_connect_task = asyncio.create_task(self._multi_try_connect(uri))
+            for task in asyncio.as_completed([self.try_connect_task], timeout=timeout):
+                await task
+        except asyncio.CancelledError:
+            raise AbortingNodeConnection
```

Comment:
Let's not overengineer these before it's actually needed for something.
```suggestion
        self.try_connect_task = asyncio.create_task(self._multi_try_connect(uri))
        try:
            await asyncio.wait_for(self.try_connect_task, timeout=timeout)
        except asyncio.CancelledError:
            raise AbortingNodeConnection
```